security/pkg/credentialfetcher/plugin/gce.go

```diff
@@ -80,5 +84,5 @@ func (p *GCEPlugin) GetType() string {
 // GetIdentityProvider returns the name of the identity provider that can authenticate the workload credential.
 // GCE idenity provider is named "GoogleComputeEngine".
```

Comment:
Why would the GCE credential fetcher ever not have GoogleComputeEngine as the IDP?

Reply:
sometimes we support the other credential fetcher like AWS. so we need to enable the customer to set the customer identity provider when setup vm

Reply:
Can you explain what it means to have
```
CredentialFetcher=AWS
IDP=GCE
```
?

Reply:
IDP is used in IDNS mapping. and when CredentialFetcher=AWS, IDP should also be AWS

Reply:
if CredentialFetcher must always match IDP then why is it configurable?

Reply:
user can set multiple identity provider even if VM is in AWS.

Reply:
Can you please explain why and how a user is expected to interact with this? If I cannot figure it out, our users will not either

Reply:
the identity provider can authenticate the workload credential and example for GCE idenity provider is named "GoogleComputeEngine".

Reply:
@howardjohn  this is required for future VM support for different IDNS mapping for different platform

Reply:
you can reference to this issue: 

Currently we hardcoded the Identity provider(GoogleComputeEngine) which is the identifier to find the correct IDNS mapping and credential Fetcher when setting up the VM. To enable the customed identity to be used in the, we choose to add a new environment variable CREDENTIAL_IDENTITY_PROVIDER to transmit this identity to enable the istio agent to find :

```
credIdentityProvider = env.RegisterStringVar("CREDENTIAL_IDENTITY_PROVIDER", "GoogleComputeEngine",
"The identity provider for CREDENTIAL. Currently Default supported identity provider is GoogleComputeEngine").Get()
```

Reply:
GCE or AWS is probably not a good example. I am thinking of on-prem VM use case, where I could have a credential fetcher reading a token from a local file, which has "filereader" CredentialFetcher type. All my on-prem VMs may be configured with the same "filereader" credentialFetcher type, but the VMs may use different IdentityProviders which have different public key endpoints. 

Credential fetcher decides how a credential is fetched for a workload. IdentityProvider specifies how the credential is authenticated. They serve different purposes. In many cases, the two match. But sometimes they are different. 

Does it make sense?